File: frontend/src/services/paths.ts
/**
 * Paths Service
 * Service for getting education pathways for careers
 */
import apiClient from './apiClient';
import { BaseResponse } from './types';

/**
 * Cost range for a pathway
 */
export interface CostRange {
  min: number;
  max: number;
  currency: string;
  description?: string;
}

/**
 * Time range for a pathway
 */
export interface TimeRange {
  min_months: number;
  max_months: number;
  description?: string;
}

/**
 * Education pathway
 */
export interface EducationPathway {
  name: string;
  cost_range: CostRange;
  time_range: TimeRange;
  pros: string[];
  tradeoffs: string[];
  description: string;
}

/**
 * Career information
 */
export interface CareerInfo {
  career_id: string;
  name: string;
  soc_code?: string;
}

/**
 * Education pathways response
 */
export interface EducationPathways {
  career: CareerInfo;
  pathways: EducationPathway[];
  available: boolean;
}

/**
 * GET /api/paths/{career_id} - Get education pathways
 * Returns 3-5 education pathways for a specific career, each with cost, time, pros, and tradeoffs
 */
export const getEducationPathways = async (career_id: string): Promise<EducationPathways> => {
  const response = await apiClient.get<BaseResponse<EducationPathways>>(
    `/api/paths/${career_id}`
  );

  if (!response.data.success) {
    throw new Error(response.data.message || 'Failed to get education pathways');
  }

  return response.data.data!;
};

/**
 * Paths service object with all methods
 */
export const pathsService = {
  getEducationPathways,
};

export default pathsService;

File: backend/routes/paths.py
"""
API routes for education pathways
"""
from fastapi import APIRouter, HTTPException, status
from models.schemas import BaseResponse, ErrorResponse
from services.paths_service import PathsService

router = APIRouter()
paths_service = PathsService()


@router.get("/{career_id}", response_model=BaseResponse)
async def get_education_paths(career_id: str):
    """
    Get education pathways for a specific career
    
    Returns 3-5 pathways, each with:
    - name: Pathway name (e.g., "Traditional 4-Year Degree", "Bootcamp + Experience")
    - cost_range: Dictionary with min, max, currency, and description
    - time_range: Dictionary with min_months, max_months, and description
    - pros: List of advantages
    - tradeoffs: List of tradeoffs/downsides
    - description: Brief overview of the pathway
    """
    try:
        result = paths_service.get_education_paths(career_id)
        
        if "error" in result:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail=ErrorResponse(
                    success=False,
                    message=result["error"]
                ).model_dump()
            )
        
        # Structure response according to requirements
        paths_data = {
            "career": result.get("career", {}),
            "pathways": result.get("pathways", []),
            "available": result.get("available", False)
        }
        
        return BaseResponse(
            success=True,
            message=f"Generated {len(paths_data['pathways'])} education pathways",
            data=paths_data
        )
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=ErrorResponse(
                success=False,
                message="Failed to generate education pathways",
                error=str(e)
            ).model_dump()
        )







